libsolidity/formal/SMTEncoder.cpp

```diff
@@ -867,6 +870,105 @@ void SMTEncoder::endVisit(ElementaryTypeNameExpression const& _typeName)
 	m_context.createExpression(_typeName, result.second);
 }
 
+namespace
+{
+unsigned dimensionCount(TypePointer _type)
+{
+	unsigned dimensionCount = 0;
+	while (_type->category() == Type::Category::Array || _type->category() == Type::Category::Mapping)
+	{
+		++dimensionCount;
+		if (_type->category() == Type::Category::Array)
+			_type = dynamic_cast<ArrayType const*>(_type)->baseType();
+		else if (_type->category() == Type::Category::Mapping)
+			_type = dynamic_cast<MappingType const*>(_type)->valueType();
+		else
+			solAssert(false, "");
+	}
+	return dimensionCount;
+}
+
+bool isReturnedFromStruct(TypePointer _type)
+{
+	// So far it seems that only Mappings and ordinary Arrays are not returned (from the cases the compiles).
+	auto category = _type->category();
+	if (category == Type::Category::Mapping)
+		return false;
+	if (category == Type::Category::Array)
+		return static_cast<ArrayType const*>(_type)->isByteArray();
+	// default
+	return true;
+}
+
+vector<string> structGetterReturnedMembers(StructType const* _structType)
+{
+	vector<string> returnedMembers;
+	for (auto const& member: _structType->nativeMembers(nullptr))
+		if (isReturnedFromStruct(member.type))
+			returnedMembers.push_back(member.name);
+	return returnedMembers;
+}
+
+}
+
+void SMTEncoder::visitPublicGetter(FunctionCall const& _funCall)
+{
+	MemberAccess const& access = dynamic_cast<MemberAccess const&>(_funCall.expression());
+	auto var = dynamic_cast<VariableDeclaration const*>(access.annotation().referencedDeclaration);
+	solAssert(var, "");
+	solAssert(m_context.knownExpression(_funCall), "");
+	auto symbArguments = applyMap(_funCall.arguments(), [this](auto argExpr) { return expr(*argExpr); });
```

Comment:
```suggestion
	auto symbArguments = applyMap(_funCall.arguments(), [this](auto argExpr) { return expr(*argExpr, parameterType); });
```
I think we should add the target type for the expression, in case conversion is needed. This might be the case for things like
```
mapping (bytes16 => uint) public m;
...
this.m("cafe");
```
or something like that.
Can you also add a test like that to see what's happening?

For the `parameterType` above, you can call `FunctionType(*var).returnParameterTypes()`

Reply:
Good catch! Without the type conversion, this crashes now for the test you suggested. I will fix it and add the test case.